PowerEditor/src/NppIO.cpp

```diff
@@ -611,7 +611,10 @@ void Notepad_plus::doClose(BufferID id, int whichOne, bool doDeleteBackup)
 	scnN.nmhdr.idFrom = (uptr_t)id;
 	_pluginsManager.notify(&scnN);
 
-	//add to recent files if its an existing file
+	// Add to recent file history only if file is removed from all the views
+	// There might be cases when file is cloned/moved to view.
+	// Don't add to recent list unless file is removed from all the views
+	generic_string fileFullPath(TEXT(""));

```

Comment:
OK, I didn't see it's used in another block. In this case, you're right.
OTOH, you don't need to do `generic_string fileFullPath(TEXT(""));`
`generic_string fileFullPath;` will create an empty string.